<a href="https://colab.research.google.com/github/luizkrawiec/estatistics/blob/main/estatistica_udemy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Dados Enem 2019**

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Exploração inicial dos Dados

In [3]:
dados = pd.read_csv('/content/microdados_enem_2019_sp.csv', sep=';', encoding='iso-8859-1')

dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  ...  NU_NOTA_MT TP_LINGUA  \
0            3                 1  ...      6002.0         0   
1            3                 1  ...         NaN         1   
2            3                 1  ...      7318.0         0   
3            3                 1  ...      3932.0         1   
4            1                 1  ...         NaN         1   

   TP_STATUS_REDACAO NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0         160.0          200.0          180.0   
1                NaN           NaN            NaN            NaN   
2                1.0         160.0          120.0          200.0   
3                1.0          80.0          160.0          120.0   
4                NaN           NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          200.0          200.0            940.0     B  
1            NaN            NaN              NaN     A  
2          200.0          200.0            880.0     A  
3          100.0          100.0            560.0     B  
4            NaN            NaN              NaN     B  

[5 rows x 44 columns]

In [4]:
dados.shape

(813772, 44)

O número total de inscritos em São Paulo foi de 813772, sendo 15,97% de todo o Brasil que foi de 5095308 inscritos.

In [5]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 813772 entries, 0 to 813771
Data columns (total 44 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   NU_INSCRICAO             813772 non-null  object 
 1   CO_MUNICIPIO_RESIDENCIA  813772 non-null  int64  
 2   NO_MUNICIPIO_RESIDENCIA  813772 non-null  object 
 3   CO_UF_RESIDENCIA         813772 non-null  int64  
 4   SG_UF_RESIDENCIA         813772 non-null  object 
 5   NU_IDADE                 813772 non-null  int64  
 6   TP_SEXO                  813772 non-null  object 
 7   TP_ESTADO_CIVIL          813772 non-null  int64  
 8   TP_COR_RACA              813772 non-null  int64  
 9   TP_NACIONALIDADE         813772 non-null  int64  
 10  CO_MUNICIPIO_NASCIMENTO  794063 non-null  float64
 11  NO_MUNICIPIO_NASCIMENTO  794063 non-null  object 
 12  CO_UF_NASCIMENTO         794063 non-null  float64
 13  SG_UF_NASCIMENTO         794063 non-null  object 
 14  TP_S

# **Limpeza de Dados**

In [6]:
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])

In [7]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  CO_UF_RESIDENCIA SG_UF_RESIDENCIA  \
0     1,90E+11            Santa Isabel                35               SP   
1     1,90E+11             Pinhalzinho                35               SP   
2     1,90E+11               São Paulo                35               SP   
3     1,90E+11   São Bernardo do Campo                35               SP   
4     1,90E+11    São Joaquim da Barra                35               SP   

   NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  TP_COR_RACA  TP_NACIONALIDADE  \
0        22       M                1            3                 1   
1        19       M                1            3                 1   
2        17       F                1            3                 1   
3        19       M                1            3                 1   
4        37       M                2            1                 1   

   CO_MUNICIPIO_NASCIMENTO  ... NU_NOTA_MT  TP_LINGUA TP_STATUS_REDACAO  \
0                1506500.0  ...     6002.0          0               1.0   
1                1503804.0  ...        NaN          1               NaN   
2                1501402.0  ...     7318.0          0               1.0   
3                3548708.0  ...     3932.0          1               1.0   
4                3549409.0  ...        NaN          1               NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   
1            NaN            NaN            NaN            NaN            NaN   
2          160.0          120.0          200.0          200.0          200.0   
3           80.0          160.0          120.0          100.0          100.0   
4            NaN            NaN            NaN            NaN            NaN   

   NU_NOTA_REDACAO  Q025  
0            940.0     B  
1              NaN     A  
2            880.0     A  
3            560.0     B  
4              NaN     B  

[5 rows x 43 columns]

In [8]:
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO'], )

In [9]:
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC'],)

In [10]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'],)

In [11]:
dados1.shape

(813772, 29)

Enriquecendo Dados

In [18]:
dados1.loc[:,'NU_NOTA_CN'] /= 10
dados1.loc[:,'NU_NOTA_CH'] /= 10
dados1.loc[:,'NU_NOTA_LC'] /= 10
dados1.loc[:,'NU_NOTA_MT'] /= 10

In [19]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]